Jellyfin.Api/Controllers/EnvironmentController.cs

```diff
@@ -0,0 +1,206 @@
+        /// <param name="includeDirectories">An optional filter to include or exclude folders from the results. true/false.</param>
+        /// <response code="200">Directory contents returned.</response>
+        /// <returns>Directory contents.</returns>
+        [HttpGet("DirectoryContents")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public IEnumerable<FileSystemEntryInfo> GetDirectoryContents(
+            [FromQuery, BindRequired] string path,
+            [FromQuery] bool includeFiles,
+            [FromQuery] bool includeDirectories)
+        {
+            const string networkPrefix = UncSeparatorString + UncSeparatorString;
+            if (path.StartsWith(networkPrefix, StringComparison.OrdinalIgnoreCase)
+                && path.LastIndexOf(UncSeparator) == 1)
+            {
+                return Array.Empty<FileSystemEntryInfo>();
+            }
+
+            var entries = _fileSystem.GetFileSystemEntries(path).OrderBy(i => i.FullName).Where(i =>
+            {
+                var isDirectory = i.IsDirectory;
+
+                if (!includeFiles && !isDirectory)
+                {
+                    return false;
+                }
+
+                return includeDirectories || !isDirectory;
+            });
+
+            return entries.Select(f => new FileSystemEntryInfo
+            {
+                Name = f.Name,
+                Path = f.FullName,
+                Type = f.IsDirectory ? FileSystemEntryType.Directory : FileSystemEntryType.File
+            });
+        }
+
+        /// <summary>
+        /// Validates path.
+        /// </summary>
+        /// <param name="validatePathDto">Validate request object.</param>
+        /// <response code="200">Path validated.</response>
+        /// <response code="404">Path not found.</response>
+        /// <returns>Validation status.</returns>
+        [HttpPost("ValidatePath")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public ActionResult ValidatePath([FromBody, BindRequired] ValidatePathDto validatePathDto)
+        {
+            if (validatePathDto.IsFile.HasValue)
+            {
+                if (validatePathDto.IsFile.Value)
+                {
+                    if (!System.IO.File.Exists(validatePathDto.Path))
+                    {
+                        return NotFound();
+                    }
+                }
+                else
+                {
+                    if (!Directory.Exists(validatePathDto.Path))
+                    {
+                        return NotFound();
+                    }
+                }
+            }
+            else
+            {
+                if (!System.IO.File.Exists(validatePathDto.Path) && !Directory.Exists(validatePathDto.Path))
+                {
+                    return NotFound();
+                }
+
+                if (validatePathDto.ValidateWritable)
```

Comment:
wtf is this doing, why is it only done when `IsFile` is null, and why is it done if `dto.Path` points to a file?

Reply:
I think the goal was to determine if a path was writable. Once we start retooling the api functionality we can make possibly breaking changes